kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/NetworkConfigurationServicePropertiesBuilder.java

```diff
@@ -258,8 +258,10 @@ private void setModemProperties() {
         if (this.gwtConfig instanceof GwtModemInterfaceConfig) {
             GwtModemInterfaceConfig gwtModemConfig = (GwtModemInterfaceConfig) this.gwtConfig;
 
-            this.properties.setModemAuthType(this.ifname, Optional.ofNullable(gwtModemConfig.getAuthType().name()));
-            this.properties.setModemPdpType(this.ifname, Optional.ofNullable(gwtModemConfig.getPdpType().name()));
+            this.properties.setModemAuthType(this.ifname,
+                    EnumsParser.getAuthType(Optional.ofNullable(gwtModemConfig.getAuthType().name())));
```

Comment:
`AuthType.name()` should never return `null`, can `gwtModemConfig.getAuthType()` return `null`? If yes probably we should do `Optional.ofNullable(gwtModemConfig.getAuthType()).map(AuthType::name)`